core/silkworm/common/util.cpp

```diff
@@ -24,13 +24,11 @@
 
 namespace silkworm {
 
-ByteView left_pad(ByteView view, size_t min_size) {
+ByteView left_pad(ByteView view, size_t min_size, Bytes& padded) {
     if (view.size() >= min_size) {
         return view;
     }
 
-    thread_local Bytes padded;
-
     if (padded.size() < min_size) {
         padded.resize(min_size);
     }
```

Comment:
Don't understand this conditional resize 
I would go for 
```suggestion
    padded.resize(view.size());                             // Resize buffer to size of input view
    size_t left_chars{min_size - view.size()};              // Compute number of chars to the left
    std::memmove(padded.data(), view.data(), view.size());  // Copy data
    padded.insert(0, left_chars, '\0');                     // Insert zeroes
    return padded;
```

Reply:
right_pad has also (imho) unnecessary double resize

Reply:
Since repeated padding is allowed, the input view might actually be a view of the buffer (or its substring), so I take care not to destroy any content of the buffer prior to the memmove. I've added comments.

Reply:
Got it now but I have yet a couple of questions

1st Looking at left_pad ...
```
    if (view.size() >= min_size) {
        return view;
    }

    if (buffer.size() < min_size) {
        buffer.resize(min_size);
    } else {
        // view & buffer might overlap in memory,
        // so we avoid shrinking the buffer prior to the memmove
    }

    assert(view.size() < min_size); // <--- This assertion can never fail due to the return on top
    size_t prefix_len{min_size - view.size()};
    [ ... ]
```

2nd. Why don't keep the original implementation keeping `buffer` as `static` (instead of thread_static) ? This way the implementation would require always a `memcpy` (impossible to have view as view-of-buffer) and original buffer gets never modified.

Sorry for pedancy

Reply:
1. Yep, that assert can never fail, it's just there for extra clarity
2. `static` would be unsafe in a multi-threaded environment (e.g. simultaneous calls to left_pad from different threads).

Reply:
1. Ok ... I find "confusing" an assertion not needed. Anyway not a big deal
2. So I understand `core` does not support natively multi-threading but **can** be used as library in higher level multi-threaded code.

Goctha now.